src/qml/filters/gpstext/ui.qml

```diff
@@ -0,0 +1,815 @@
+			filter.set('style', 'normal')
+			filter.set(textFilterUi.useFontSizeProperty, false)
+			filter.set('size', profile.height)
+			
+			// Add default preset.
+			filter.set(textFilterUi.valignProperty, 'bottom')
+			filter.set(textFilterUi.halignProperty, 'left')
+			filter.set(textFilterUi.rectProperty, '10%/10%:80%x80%')
+			filter.savePreset(presetParams)
+		} else {
+			filter.set(textFilterUi.middleValue, filter.getRect(textFilterUi.rectProperty, filter.animateIn + 1))
+			if (filter.animateIn > 0)
+				filter.set(textFilterUi.startValue, filter.getRect(textFilterUi.rectProperty, 0))
+			if (filter.animateOut > 0)
+				filter.set(textFilterUi.endValue, filter.getRect(textFilterUi.rectProperty, filter.duration - 1))
+		}
+		filter.blockSignals = false
+		setControls()
+		
+		if (filter.isNew)
+			filter.set(textFilterUi.rectProperty, filter.getRect(textFilterUi.rectProperty))
+			
+		if (gpsFile.exists()) {
+			fileLabel.text = gpsFile.fileName
+			fileLabelTip.text = gpsFile.filePath
+		} else {
+			fileLabel.text = qsTr("No File Loaded")
+			fileLabel.color = 'red'
+			fileLabelTip.text = qsTr('No GPS file loaded.\nClick "Open" to load a file.')
+		}
+	}
+
+	FileDialog {
+		id: fileDialog
+		modality: application.dialogModality
+		selectMultiple: false
+		selectFolder: false
+		folder: settingsOpenPath
+		nameFilters: ['Supported files (*.gpx *.tcx)', 'GPS Exchange Format (*.gpx)', 'Training Center XML (*.tcx)']
+		onAccepted: {
+			gpsFile.url = fileDialog.fileUrl
+			gpsTextRoot.fileOpened(gpsFile.path)
+			fileLabel.text = gpsFile.fileName
+			fileLabel.color = activePalette.text
+			fileLabelTip.text = gpsFile.filePath
+			filter.set('gps.file', gpsFile.url)
+			gpsFinishParseTimer.restart()
+			filter.set('gps_start_text', '')
+			filter.set('gps_processing_start_time', 'yyyy-MM-dd hh:mm:ss');
+		}
+	}
+	
+	function smooth_value_to_index(val) {
+		switch (parseInt(val)) {
+			case 0: return 0;
+			case 1: return 1;
+			case 3: return 2;
+			case 5: return 3;
+			case 7: return 4;
+			case 11: return 5;
+			case 15: return 6;
+			case 31: return 7;
+			case 63: return 8;
+			case 127: return 9;
+            default: {
+                console.log("default switch, val= " + val);
+                return 0;
+            }
+		}
+	}
+	
+	//timer to update UI after gps file is processed; max: 10x250ms
+	Timer {
+		id: gpsFinishParseTimer
+		interval: 250
+		repeat: true
+		triggeredOnStart: false
+		property int calls: 0
+		onTriggered: {
+			if (filter.get('gps_start_text') == '' || filter.get('gps_start_text') == '--') {
+				calls += 1;
+				if (calls > 10) {
+					gpsFinishParseTimer.stop();
+					setControls();
+					calls = 0;
+				}
+			}
+			else {
+				gpsFinishParseTimer.stop();
+				calls = 0;
+				setControls();
+			}
+		}
+	}
+
+    //set all textbox values at start or on refresh
+	function setControls() {
+		textArea.text = filter.get('argument');
+		textFilterUi.setControls();
+		
+		if (filter.isNew) {
```

Comment:
All these filter.set() calls should be moved to Component.onCompleted. setControls should not change values on the filter.

Reply:
Separated everything properly in latest commit.